src/hotspot/share/opto/node.cpp

```diff
@@ -974,6 +974,18 @@ Node* Node::find_out_with(int opcode) {
   return nullptr;
 }
 
+Node* Node::find_unique_out_with(int opcode) const {
```

Comment:
Random idea:
Would it not be nice if this method automatically casted the node to that node-class?
Suggestions:
- using templates: give the class name and the opcode. A bit annoying to use
- using macros: give it the node-type name: i.e. `Add` for `AddNode`. The macro then uses the template, filling in `AddNode` and `Op_Add`. What do you think?

Reply:
Yes, it would but that out of scope for this PR.